File: content/examples/other/leetcode/problems/container-with-most-water/example-1/Solution.java
public class Solution {
  static int maxAreaBrute(int[] height) {
    int best = 0;
    for (int i = 0; i < height.length; i += 1) {
      for (int j = i + 1; j < height.length; j += 1) {
        int area = Math.min(height[i], height[j]) * (j - i);
        if (area > best) best = area;
      }
    }
    return best;
  }

  public static void main(String[] args) {
    System.out.println(
        maxAreaBrute(new int[] {1, 8, 6, 2, 5, 4, 8, 3, 7}));
  }
}

File: content/examples/other/leetcode/problems/container-with-most-water/example-1/solution.js
function maxAreaBrute(height) {
  let best = 0;
  for (let i = 0; i < height.length; i += 1) {
    for (let j = i + 1; j < height.length; j += 1) {
      const area = Math.min(height[i], height[j]) * (j - i);
      if (area > best) best = area;
    }
  }
  return best;
}

if (require.main === module) {
  console.log(maxAreaBrute([1, 8, 6, 2, 5, 4, 8, 3, 7]));
}

module.exports = { maxAreaBrute };
